### ISSUE TITLE
Please add a phonetic phoneme option to the NVDA dicspeech dictionaries dialogue

### ISSUE THREAD

**Issue by M3198** [nvaccess/nvda#20487]
### Is your feature request related to a problem? Please describe.

Yes, it is. I'm using IBM tts with NVDA 2026.1.1. The issue is, since IBM tts uses the eci engine rule, some words are corrected through phoneme rules rather than standard spellings. I tried to use regular expression, however, rather than reading it as it should, it reads it as a regular string of code.
I don't have this issue with the codefactory sapi5 eloquence engine.
The word in question is beknown
The replacement is
`[.0bX.1non]

### Describe the solution you'd like

The solution would be to add a phonetic phoneme rule to the list of type of dictionary entries. In the current NVDA release, you only have whole word, anywhere, and regular expression, neither of which have worked. I even tried the wildcards option in the NVDA 2026.2 beta, it doesn't fix the issue.

### Describe alternatives you've considered

There is no other alternative that works. I've even tried to edit the dictionary file manually, but it hasn't fixed the issue at all.

### Additional context

This should be for the voice dictionary option in NVDA, preferences, speech dictionaries submenu.

**Comment by seanbudd** [comment]
Closing as won't fix. We cannot control how synthesizers pronounce words.

**Comment by CyrilleB79** [comment]
@seanbudd I disagree with this reason for closing.

Some synth can pronounce phonemes when instructed to, and NVDA has actually a `PhonemeCommand` speech command; though for now, it only supports the "th" sound and was used by MathPlayer to speak "4th", "13th", etc. At least one of NVDA's natively supported synth (OneCore or eSpeak) supports `PhonemeCommands` too IIRC. Regarding the phoneme syntax, NVDA uses IPA notation for `PhonemeCommand`s so in any case, IBMTTS should update its driver to convert NVDA's `PhonemeCommand`'s IPA to its own specific notation.

It would be completely possible for NVDA to support other phonemes and add a dedicated dictionary entry type. If we want to is an open question, but again, if you stick with closing this issue, another reason should be invoked for it.

**Comment by M3198** [comment]
Exactly. This issue needs to be reopened and looked at thoroughly.

It is quite unfair that legitimate issues like this get closed down quite abruptly. Nobody is saying that NVDA should control how each and every synth should pronounce every word. However users should have the right to have words pronounced as they wish, which is why we have the speech dictionaries in the first place. So I would say this issue must be considered.

**Comment by CyrilleB79** [comment]
Correction:
NVDA should actually support `PhonemeCommand`s for all phonemes, provided the synth driver supports it.

Actually, OneCore seems to be able to produce standard phonemes, not only a tiny subset. The following example in Python console should work:
```
from speech.commands impor PhonemeCommand
speech.speak([PhonemeCommand("həˈləʊ ˌɛn ˈviː ˈæk.sɛs")])
```

I have not tested because I have no English OneCore installed here, but with the following French string, it works with French OneCore:
`speech.speak([PhonemeCommand("bɔ̃.ʒuʁ ˌɛn ˈviː ˈæk.sɛs")])`

`eSpeak` on the opposite, supports only 2 or 3 phonemes to be able to speak the ones used in math reading.

Also, seems that MathCAT too use the "th" phoneme, not only the now deprecated MathPlayer.

**Comment by makhlwf** [comment]
> Actually, OneCore seems to be able to produce standard phonemes, not only a tiny subset. The following example in Python console should work:
> 
> ```
> from speech.commands impor PhonemeCommand
> speech.speak([PhonemeCommand("həˈləʊ ˌɛn ˈviː ˈæk.sɛs")])
> ```
> 
> I have not tested because I have no English OneCore installed here

i have tryed it and i can conferm it works with onecore english voices

**Comment by M3198** [comment]
So then it shouldn't be too difficult to implement, because 2026.2 will be implementing custom dictionary support anyway?

**Comment by CyrilleB79** [comment]
No, nothing to do with 2026.2's custom dictionaries.

**Comment by M3198** [comment]
Okay but we should still bring this option to NVDA. It shouldn't be too difficult to do.

**Comment by SaschaCowley** [comment]
When implemented, this will likely be based around Unicode IPA transcriptions, and will only be compatible with speech synthesizers which are capable of interpreting same.

**Comment by M3198** [comment]
Hello,

This phoneme feature should support other phoneme codes, such as the eci phoneme engine. Of course, implementing phoneme codes for every single synth might be too difficult, so perhaps in the NVDA preferences speech settings there could be a checkbox called enable custom phonemes for custom synths?
Best regards,
Muhammad.

**Comment by CyrilleB79** [comment]
No @M3198, NVDA's `PhonemeCommand`s support and should continue supporting IPA. If you want other synth systems to be supported,it's the responsibility of the specific drivers for these items. For example drivers for eci synth such as IBMTTS or Eloquence should support translating the IPA in `PhonemeCommand` to eci phoneme syntax.

**Comment by Mohamed00** [comment]
IBMTTS also supports loading the built-in abbreviations, main and roots dictionaries provided by the synthesizer.
To add this word to the dictionary, create a file named enuroot.dic and place in it the following contents:
> beknown	`[.0bX.1non]

Replace the space with a tab.
Then place this file in the location of your IBMTTS libraries, and restart NVDA or reload the synthesizer.

**Comment by M3198** [comment]
Thanks it worked!

IBM tts synth drivers ibmtts. Pasted it. Thank you very much!

**Comment by CyrilleB79** [comment]
@seanbudd shouldn't we reopen this issue? Even if the op has found a solution for IBMTTS, the solutions triaged here would be for any synth.

**Comment by M3198** [comment]
I agree. It was pretty premature of me to close it as soon as I found a solution, and apologies if I've caused any inconvenience.

**Comment by LeonarddeR** [comment]
I have no opinion other than that we should use a standardized approach (IPA) indeed. I don't think we should add just another type, rather use an extra check box that allows including IPA. The syntax to include the IPA phonemes should be compatible with regex replacements.
